extensions/avro/src/main/java/com/hazelcast/jet/avro/impl/AvroSerializerHooks.java

```diff
@@ -156,6 +141,20 @@ public GenericContainer read(@Nonnull ObjectDataInput in) throws IOException {
         public boolean isOverwritable() {
             return false;
         }
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance instance) {
+            if (instance != null) {
+                instance.getLifecycleService().addLifecycleListener(event -> {
+                    if (event.getState() == STARTED) {
+                        ((AbstractSerializationService) ((SerializationServiceSupport) instance)
+                                .getSerializationService()).register(
+                                        SpecificRecordBase.class,
```

Comment:
This is a very convoluted way to register the serializer. However, you cannot simply have a hook with getSerializationType=SpecificRecordBase and createSerializer=ExternalizableSerializer, because ExternalizableSerializer must be configured (in particular class filter).

Reply:
and is should be simpler to use AbstractSerializationService.registerFromSuperType instead of registering by id

Reply:
> alternatively you might add default createSerializer(SerializationService) method to SerializerHook

Can we mark both `createSerializer()` and `createSerializer(SerializationService)` as default, so that when I implement one, I can skip the other?

Reply:
you do not need that, and you cannot invoke `createSerializer(SerializationService)` from `createSerializer` - serialization service should be `@Nonnull`.

If `createSerializer(SerializationService)` is the only one invoked externally (that should be the case), then you can override only 1 method to get desired behavior

Reply:
I agree, if there's an easy way to register the serializer, let's do that.

Reply:
I switched to using `registerFromSuperType`, but actually even `toData` will trigger registration. WDYT? The users will depend on it for the same functionality.

Reply:
> but actually even toData will trigger registration. WDYT? 

`toData` registers a type so it does not have to traverse the superclasses and implemented interfaces (performance optimization for lazy registration, also this way it is less dependent on the order of registrations). We do it in `GenericContainerHook` to ensure that the correct serializer is used (the one for `Externalizable`, not for `GenericRecord`).

> The users will depend on it for the same functionality. We may need to expose registerFromSuperType to users at some point since it is more flexible than toData.

Registration in toData is only lazy/performance. Currently there is no need to expose `registerFromSuperType` it. I would say even that our usage is exception rather than rule. And we only need to do it that way because `ExternalizableSerializer` has configuration. If it didn't have it, we could simply create a new instance and return it from the hook.